docs/source/advanced/contrib_thunder.rst

```diff
@@ -0,0 +1,468 @@
+The core language elementwise binary operations, however, do perform broadcasting and type promotion.
+
+Before adding a core language operation consider if the functionality expressed is universal enough.
+
+As a style note, operations in *Thunder* should defer as much error checking as possible.
+For example, if a primitive’s meta function will perform an error check for ``X``,
+then the core language operation that calls it should generally not also check for ``X``.
+
+
+------------------
+The Torch Language
+------------------
+To translate ``torch`` operations into something that *Thunder* understands we define a
+`torch language <https://github.com/Lightning-AI/lightning-thunder/blob/main/thunder/torch/__init__.py>`_.
+Operations in the ``torch`` language should be written as if a practitioner were writing them,
+and they should reflect the behavior of their corresponding torch operations (small deviations are sometimes OK).
+
+When a program is interpreted, torch operations are remapped into these operations, which ultimately call primitive operations.
+
+
+----------------
+Language Context
+----------------
+In the ``core`` and ``torch`` languages functions are decorated to set a *language context* and – for ``torch`` operations – to describe
+how to map operations like ``torch.foo`` into ``thunder.torch.foo``.
+
+The language context determines what properties and methods tensor objects have.
+For example, when ``a + b`` is written, the language context decides what that addition means.
+Or when ``a.size`` is used, the language context determines what that means (and it’s different in PyTorch and NumPy).
+
+
+---------------------------------------
+Adding operations to the Torch executor
+---------------------------------------
+Now that we are familiar with the hierarchy of operations and the underlying language contexts, let's see some
+examples of adding operations.
+
+For simplicity, we only cover adding operations to the ``torch`` executor.
+The sections below are meant to be read **sequentially**.
+
+~~~~~~~~~~~~~~~~~~
+Adding a primitive
+~~~~~~~~~~~~~~~~~~
+A good example of adding a primitive operation to the ``torch`` executor is
+the `PR #136 <https://github.com/Lightning-AI/lightning-thunder/pull/136>`_
+which adds support for `torch.Tensor.unfold <https://pytorch.org/docs/stable/generated/torch.Tensor.unfold.html>`_.
+
+Let's outline some of its key parts.
+
+The following added (emphasized) lines to
+`thunder/core/prims.py <https://github.com/Lightning-AI/lightning-thunder/blob/main/thunder/core/prims.py>`_
+
+.. code-block:: python
+   :emphasize-lines: 5
+   :lineno-start: 152
+
+   SLICE = auto()
+   SQUEEZE = auto()
+   TRANSPOSE = auto()
+   UNFOLD = auto()
+   VIEW = auto()
+   # Memory layout prims (Experimental)
+   STRIDE_ORDER = auto()
+
+.. code-block:: python
+   :emphasize-lines: 1-18
```

Comment:
This is highlightning the entire snippet

Reply:
So, I used it to indicate the added by the contributor lines. It is possible to get that highlighted similar to how git shows that. But I would need to manipulate Sphinx a bit more, not an expert there. What do you think, @carmocca . Or do you think that dropping any highlight is better?

Reply:
Dropping it is probably the best option in the cases where it's the whole snippet. If you want to highlight only a subset then I would keep it.